src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt

```diff
@@ -0,0 +1,462 @@
+      return@map caCase.copy(
+        probationPractitioner = ProbationPractitioner(
+          staffCode = caCase.probationPractitioner?.staffCode,
+          name = caCase.probationPractitioner?.name,
+        ),
+      )
+    }
+
+    return caCaseList
+  }
+
+  private fun buildCaseload(cases: List<CaCase>, searchString: String?, view: String): CaCaseLoad {
+    val showAttentionNeededTab = cases.any { it.tabType == CaViewCasesTab.ATTENTION_NEEDED }
+    val searchResults = applySearch(searchString, cases)
+    val sortResults = applySort(searchResults, view)
+    return CaCaseLoad(
+      cases = sortResults,
+      showAttentionNeededTab,
+    )
+  }
+
+  private fun applySearch(searchString: String?, cases: List<CaCase>): List<CaCase> {
+    if (searchString == null) {
+      return cases
+    }
+    val term = searchString.lowercase()
+    return cases.filter {
+      it.name.lowercase().contains(term) ||
+        it.prisonerNumber.lowercase().contains(term) ||
+        it.probationPractitioner?.name?.lowercase()?.contains(term) ?: false
+    }
+  }
+
+  private fun applySort(cases: List<CaCase>, view: String): List<CaCase> =
+    if (view == "prison") cases.sortedBy { it.releaseDate } else cases.sortedByDescending { it.releaseDate }
+
+  private fun filterAndFormatExistingLicences(licences: List<LicenceSummary>): List<CaCase> {
+    val preReleaseLicences = licences.filter { it.licenceStatus != ACTIVE }
+    if (preReleaseLicences.isEmpty()) {
+      return emptyList()
+    }
+
+    val licenceNomisIds = preReleaseLicences.map { it.nomisId }
+    val prisonersWithLicences = caseloadService.getPrisonersByNumber(licenceNomisIds)
+    val nomisEnrichedLicences = this.enrichWithNomisData(preReleaseLicences, prisonersWithLicences)
+    return filterExistingLicencesForEligibility(nomisEnrichedLicences)
+  }
+
+  private fun findAndFormatNotStartedLicences(
+    licences: List<LicenceSummary>,
+    prisonCaseload: List<String>,
+  ): List<CaCase> {
+    val licenceNomisIds = licences.map { it.nomisId }
+    val prisonersApproachingRelease = getPrisonersApproachingRelease(prisonCaseload)
+
+    val prisonersWithoutLicences = prisonersApproachingRelease.filter { p ->
+      !licenceNomisIds.contains(p.prisoner.prisonerNumber)
+    }
+    val eligiblePrisoners = filterOffendersEligibleForLicence(prisonersWithoutLicences.toList())
+    val casesWithoutLicences = pairNomisRecordsWithDelius(eligiblePrisoners)
+    return createNotStartedLicenceForCase(casesWithoutLicences)
+  }
+
+  private fun createNotStartedLicenceForCase(cases: List<ManagedCase>): List<CaCase> = cases.map { c ->
+
+    // Default status (if not overridden below) will show the case as clickable on case lists
+    var licenceStatus = NOT_STARTED
+
+    if (isBreachOfTopUpSupervision(c)) {
+      // Imprisonment status indicates a breach of top up supervision order - not clickable (yet)
+      licenceStatus = OOS_BOTUS
+    } else if (isRecall(c)) {
```

Comment:
note: Not using the existing isRecallCase method in eligibility service as it is not returning false if recall flag is null.
Also it's a bit strange as we have to do all checks(**getIneligibilityReasons**) in order to execute **isRecallCase**.